spec/controllers/posts_controller_spec.rb

```diff
@@ -1,51 +1,29 @@
 require 'rails_helper'
 
-describe UsersController, type: :request do
+describe Post, type: :request do
   before do
     @user = User.create(name: 'Sam', photo: 'photo.url', bio: 'Excellent', posts_counter: 0)
+    @post = Post.create(author: @user, title: 'Let us go', text: 'This is the text for the post')
   end
 
-  describe 'GET #index' do
-    it 'responds with success' do
-      get '/users'
-      expect(response).to have_http_status(:success)
-    end
-
-    it 'renders the index template' do
-      get '/users'
-      expect(response).to render_template('users/index')
+  describe 'routes with post_controller' do
+    it 'should render the index action correctly' do
+      get "/users/#{@user.id}/posts"
+      expect(response).to be_successful
+      expect(response).to render_template('posts/index')
       expect(response).to render_template(layout: 'layouts/application')
     end
 
-    it 'assigns @users with all users' do
-      get '/users'
-      expect(assigns(:users)).to eq(User.all)
-    end
-
-    it 'includes the correct placeholder text in the response body' do
-      get '/users'
-      expect(response.body).to include('Names of all the users')
-      expect(response.body).to include('Name: Sam')
-      expect(response.body).to include('Posts Counter: 0')
-      expect(response.body).to include('Posts: #&lt;ActiveRecord::Associations::CollectionProxy []&gt;')
-    end
-  end
-
-  describe 'GET #show' do
-    it 'responds with success' do
-      get "/users/#{@user.id}"
-      expect(response).to have_http_status(:success)
-    end
-
-    it 'renders the show template' do
-      get "/users/#{@user.id}"
-      expect(response).to render_template('users/show')
-      expect(response).to render_template(layout: 'layouts/application')
+    it 'should render the index content correctly' do
+      get "/users/#{@user.id}/posts"
+      expect(response.body).to include('user page')
     end
 
-    it 'includes the correct placeholder text in the response body' do
-      get "/users/#{@user.id}"
-      expect(response.body).to include('this is the detail page for the user with id')
+    it 'should render the show action correctly' do
+      get "/users/#{@user.id}/posts/1"
+      expect(response).to be_successful
+      expect(response).to render_template('posts/show', layout: 'layouts/application')
+      expect(response.body).to include('this is your detail page')
     end
```

Comment:
- [ **OPTIONAL** ] Good job adding changes mentioned by the previous reviewer. However, I think it would be better if you all tests about the `show` action in `describe` method and for each single test in an `it` method. this will make your code looks more readable, organized and professional. You can apply this approach to the whole file. For example:
```ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  it 'should render the show action correctly' do
    ....
  end

  it 'should test If the response body includes correct placeholder text.' do
    ....
  end
  ...
end
```